rfd/0167-console-service.md

```diff
@@ -0,0 +1,194 @@
+---
+authors: Gabriel Corado (gabriel.oliveira@goteleport.com)
+state: draft
+---
+
+# RFD 0167 - Console service
+
+## Required Approvals
+
+* Engineering: @r0mant && @greedy52
+* Security: @reedloden || @jentfoo
+
+## What
+This RFD proposes a mechanism for Teleport administrators to switch log-level
+and enable profile consumption without restarting the instance.
+
+## Why
+During incidents in production environments, debug logs and diagnostic profiles
+(such as CPU and Memory allocation) are often unavailable for timely
+troubleshooting. To enable those, administrators need to restart the instance,
+potentially disrupting the application state and obscuring the issue they're
+trying to solve.
+
+## Details
+
+A separate service will be introduced specifically for troubleshooting
+purposes. This will ensure the current diagnosis service is kept as is, not
+breaking current integrations or introducing behavior changes. The new service
+will also always be available (contrasting with the diagnosis service, which
+is optional and disabled by default), so in scenarios where users need to use
+it, they don't need to restart Teleport.
+
+In addition, the new service will listen into a Unix socket instead of TCP. This
+will make discoverability easier and discourage external usage, as the API is
+designed for internal usage.
+
+The `pprof` endpoints on diagnostics will still be available for existent
+integrations, such as the usage of
+`go tool pprof http://diag-addr/debug/pprof/profile`.
+
+### New service
+
+Teleport will start listening using a Unix socket located at
+`/tmp/teleport-diag-<node-id>.sock`.
+
+Having the Node ID on the socket name will also cover scenarios where multiple
+instances of running on the same machine exist. In this case, the consumers can
+rely on the Teleport configuration to locate the data directory and retrieve
+the ID.
+
+#### Endpoints
+
+In addition to the `pprof` endpoints, the service will also have an endpoint to
+change the applications's log level.
+
+`POST /set-log-level` will receive the new log level on its body as text. The
```

Comment:
`set-` is redundant as HTTP method is used for the action type. And in this case I believe `PUT` is more appropriate than `POST`.

Maybe also have a `GET` call to check the current log level?